cadl-extension/src/main/java/com/azure/cadl/Main.java

```diff
@@ -0,0 +1,140 @@
+        LOGGER.info("Count of XML files: {}", javaPackage.getXmlFiles().size());
+        LOGGER.info("Count of text files: {}", javaPackage.getTextFiles().size());
+
+        // write output
+        Formatter formatter = new Formatter();
+        for (JavaFile javaFile : javaPackage.getJavaFiles()) {
+            String content = javaFile.getContents().toString();
+            try {
+                content = formatter.formatSourceAndFixImports(content);
+            } catch (Exception e) {
+                LOGGER.error("Failed to format file: {}", outputFolder + javaFile.getFilePath(), e);
+//                continue;
+            }
+            cadlPlugin.writeFile(outputFolder + javaFile.getFilePath(), content, null);
+        }
+        for (XmlFile xmlFile : javaPackage.getXmlFiles()) {
+            String content = xmlFile.getContents().toString();
+            cadlPlugin.writeFile(outputFolder + xmlFile.getFilePath(), content, null);
+        }
+        for (TextFile testFile : javaPackage.getTextFiles()) {
+            String content = testFile.getContents();
+            cadlPlugin.writeFile(outputFolder + testFile.getFilePath(), content, null);
+        }
+        // resources
+        String artifactId = ClientModelUtil.getArtifactId();
+        if (!CoreUtils.isNullOrEmpty(artifactId)) {
+            cadlPlugin.writeFile(outputFolder + "src/main/resources/" + artifactId + ".properties",
+                    "name=${project.artifactId}\nversion=${project" + ".version}\n", null);
+        }
+    }
+
+    private static CodeModel loadCodeModel(String filename) throws IOException {
+        String file = readFile(filename);
+
+        Representer representer = new Representer();
+        representer.setPropertyUtils(new AnnotatedPropertyUtils());
+        representer.getPropertyUtils().setSkipMissingProperties(true);
+        LoaderOptions loaderOptions = new LoaderOptions();
+        loaderOptions.setMaxAliasesForCollections(Integer.MAX_VALUE);
+        Constructor constructor = new CodeModelCustomConstructor(loaderOptions);
+        Yaml yamlMapper = new Yaml(constructor, representer, new DumperOptions(), loaderOptions);
+        CodeModel codeModel = yamlMapper.loadAs(file, CodeModel.class);
+        return codeModel;
+    }
+
+    private static String readFile(String path) throws IOException {
+        try (InputStream fis = new FileInputStream(path)) {
+            StringBuilder sb = new StringBuilder();
+            char[] buffer = new char[1024];
+            try (InputStreamReader reader = new InputStreamReader(fis, StandardCharsets.UTF_8)) {
+                int charsRead;
+                while ((charsRead = reader.read(buffer, 0, buffer.length)) > 0) {
+                    sb.append(buffer, 0, charsRead);
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            return sb.toString();
+        }
```

Comment:
I'd recommend transitioning this over to `new String(Files.readAllBytes(Paths.get(path)))`;